tests/python/gaia-ui-tests/gaiatest/tests/functional/settings/test_settings_change_time_format.py

```diff
@@ -0,0 +1,47 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from gaiatest import GaiaTestCase
+from gaiatest.apps.settings.app import Settings
+from gaiatest.apps.system.regions.status_bar import StatusBar
+from gaiatest.apps.lockscreen.app import LockScreen
+
+
+class TestSettingsChangeTimeFormat(GaiaTestCase):
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+
+        # Set the time on the device
+        _seconds_since_epoch = self.marionette.execute_script("""
+                    var today = new Date();
+                    var yr = today.getFullYear();
+                    var mth = today.getMonth();
+                    var day = today.getDate();
+                    return new Date(yr, mth, day, 22, 0, 0).getTime();""")
```

Comment:
Re: Robert's note, I don't think you want to count on the controlling system and the target having the same time. So doing all time operations in the context of the target makes sense here.